sdk/src/main/java/com/google/cloud/dataflow/sdk/util/TriggerRunner.java

```diff
@@ -172,6 +172,8 @@ public boolean shouldFire(W window, Timers timers, StateAccessor<?> state) throw
   }
 
   public void onFire(W window, Timers timers, StateAccessor<?> state) throws Exception {
+    // shouldFire should be false.
```

Comment:
Typo? I think `rootTrigger.invokeShouldFire(context)` should be true, otherwise it is forbidden to call `invokeOnFire`.